internal/server/config/cache_settings_test.go

```diff
@@ -0,0 +1,181 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCacheSettings(t *testing.T) {
+	t.Run("should_create_new_cache", func(t *testing.T) {
+		tests := []struct {
+			name                   string
+			cacheSettings          CacheSettings
+			expectedCreateNewCache bool
+		}{
+			{
+				name: "not_when_limit_is_zero",
+				cacheSettings: CacheSettings{
+					CheckCacheLimit: 0,
+				},
+				expectedCreateNewCache: false,
+			},
+			{
+				name: "not_when_limit_is_zero_and_query_cache_enabled",
+				cacheSettings: CacheSettings{
+					CheckCacheLimit:        0,
+					CheckQueryCacheEnabled: false,
```

Comment:
should `CheckQueryCacheEnabled` be `true`? Seems like maybe from the test `name`